File: front-end/src/app/service-models/app/v1/client/client-company-attendance-shift-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';

export class ClientCompanyAttendanceShiftSM extends CoinManagementServiceModelBase<number> {
    shiftFrom!: Date;
    shiftTo!: Date;
    shiftName!: string;
    shiftDescription!: string;
    primaryOfficeGeoCoordinatesLocation!: string;
    allowedRaidus!: string;
    locationBased!: string;
    allowedIps!: string;
    clientCompanyDetailId!: number;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/ClientCompanyAttendanceShiftSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class ClientCompanyAttendanceShiftSM : SiffrumPayrollServiceModelBase<int>
    {
        public DateTime ShiftFrom { get; set; }
        public DateTime ShiftTo { get; set; }
        public string ShiftName { get; set; }
        public string ShiftDescription { get; set; }
        public string PrimaryOfficeGeoCoordinatesLocation { get; set; }
        public string AllowedRaidus { get; set; }
        public string LocationBased { get; set; }
        public string AllowedIps { get; set; }
        public int ClientCompanyDetailId { get; set; }
    }
}
